patches/api/0327-Add-PlayerCompostItemEvent.patch

```diff
@@ -0,0 +1,112 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Sat, 7 Aug 2021 15:11:27 +0200
+Subject: [PATCH] Add PlayerCompostItemEvent.
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerCompostItemEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerCompostItemEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1195392aff2829eed6d987d1896b3fe453cafc53
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerCompostItemEvent.java
+@@ -0,0 +1,100 @@
++package com.destroystokyo.paper.event.player;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when an item is about to be composted by a player.
++ */
++public class PlayerCompostItemEvent extends PlayerEvent implements Cancellable {
```

Comment:
Non-player's can also compost items, namely villagers and non-entities that insert items into the composter like hoppers.

Reply:
What would be the best way to do this? A CompostItemEvent as parent with `EntityCompostItemEvent` and `HopperCompostItemEvent` as children?

Reply:
I don’t think you need 3 events, just at most 2 ItemCompostEvent, and EntityItemCompostEvent. You could probably just get away with one and have a nullable entity getter thats just null when it’s not coming from and entity’s action.

Reply:
> You could probably just get away with one and have a nullable entity getter thats just null when it’s not coming from and entity’s action.

I've implemented EntityEvent and overriden `getEntity` to specify this in the javadocs, is this the intended way of adding extra javadoc?